WordPress/gutenberg
# Page Content Focus: Add welcome guides

## What?
Closes https://github.com/WordPress/gutenberg/issues/51822.

Adds welcome guides which appear to the user when they open a page in the site editor and when they edit that page's template.

## How?
- New `WelcomeGuidePage` and `WelcomeGuideTemplate` components in `@wordpress/edit-site` do the bulk of the lifting.
- Supporting changes to `WelcomeGuide` in `@wordpress/components` which is what all of our guides use.
    - Use default button styling for the Previous and Next buttons instead of (buggy!) custom styling.
    - Use default accent/theme colour for Previous button, Next button, and current page indicator instead of a custom hardcoded blue.
    - Do not show X button if there is only one page.

## Testing Instructions
1. Go to Appearance → Editor → Pages and select a page to edit.
2. A guide should appear.
3. Click _Edit template_ in the sidebar.
4. Another guide should appear.
5. Check that exiting the guides in the page editor, site editor, widgets editor, and widgets customiser still work.

Dismissing the guides should prevent them from appearing again.

For testing purposes, you can run this in DevTools to re-enable the new guides:

```
wp.data.dispatch( 'core/preferences' ).toggle( 'core/edit-site', 'welcomeGuidePage' );
wp.data.dispatch( 'core/preferences' ).toggle( 'core/edit-site', 'welcomeGuideTemplate' );
```

## Screenshots or screencast 

The new guides:

https://github.com/WordPress/gutenberg/assets/612155/e1c3bedd-5ae1-4f27-a14b-0e5e0db72ce7

The old guides:

| Before | After |
| - | - | 
| <img width="1265" alt="Screenshot 2023-06-28 at 16 56 16" src="https://github.com/WordPress/gutenberg/assets/612155/171adb2a-ecdb-4d8a-ac3d-fb384e2efd2f"> | <img width="1265" alt="Screenshot 2023-06-28 at 16 55 58" src="https://github.com/WordPress/gutenberg/assets/612155/c6a5f3de-3012-463f-9482-c564cd3e9249"> |
